code/ztfm.c

```diff
@@ -0,0 +1,1432 @@
+    mps_transform_destroy(t1);
+    t1 = NULL;
+
+    /* Empty (no add) */
+    before(myrootExact, perset);
+    res = mps_transform_create(&t1, arena);
+    Insist(res == MPS_RES_OK);
+    res = mps_transform_apply(&applied, t1);
+    Insist(res == MPS_RES_OK);
+    Insist(applied);
+    t1 = NULL;
+    after(myrootExact, perset, 1, 0, 2, 0);
+
+    /* Identity-transform */
+    before(myrootExact, perset);
+    res = mps_transform_create(&t1, arena);
+    Insist(res == MPS_RES_OK);
+    for(l = k + 4; k < l; k++) {
+      mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k], 1);
+    }
+    mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k], 10);
+    k += 10;
+    res = mps_transform_apply(&applied, t1);
+    Insist(res == MPS_RES_OK);
+    Insist(applied);
+    t1 = NULL;
+    after(myrootExact, perset, 1, 0, 2, 0);
+
+    /* Mixed non-trivial, NULL- and identity-transforms */
+    before(myrootExact, perset);
+    res = mps_transform_create(&t1, arena);
+    Insist(res == MPS_RES_OK);
+    {
+      mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k + perset], 1);
+      k += 1;
+      /* NULL */
+      mps_transform_add_oldnew(t1, &nullref1, &myrootExact[k + perset], 1);
+      k += 1;
+      /* identity */
+      mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k], 1);
+      k += 1;
+      /* NULL */
+      mps_transform_add_oldnew(t1, &nullref2, &myrootExact[k + perset], 1);
+      k += 1;
+    }
+    mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k + perset], 10);
+    k += 10;
+    res = mps_transform_apply(&applied, t1);
+    Insist(res == MPS_RES_OK);
+    Insist(applied);
+    t1 = NULL;
+    after(myrootExact, perset, 1, -11, 2, +11);
+
+    /* Non-trivial transform */
+    before(myrootExact, perset);
+    res = mps_transform_create(&t1, arena);
+    Insist(res == MPS_RES_OK);
+    for(l = k + 4; k < l; k++) {
+      mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k + perset], 1);
+    }
+    mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k + perset], 10);
+    k += 10;
+    res = mps_transform_apply(&applied, t1);
+    Insist(res == MPS_RES_OK);
+    Insist(applied);
+    t1 = NULL;
+    after(myrootExact, perset, 1, -14, 2, +14);
+
+    /* Two transforms, first destroyed unused */
+    before(myrootExact, perset);
+    res = mps_transform_create(&t1, arena);
+    Insist(res == MPS_RES_OK);
+    mps_transform_add_oldnew(t1, &myrootExact[k], &myrootExact[k + perset], 10);
+    k += 10;
+    l = k;
+    res = mps_transform_create(&t2, arena);
+    Insist(res == MPS_RES_OK);
+    mps_transform_add_oldnew(t2, &myrootExact[l], &myrootExact[l + perset], 10);
+    l += 10;
+    res = mps_transform_apply(&applied, t2);
+    Insist(res == MPS_RES_OK);
+    Insist(applied);
+    t2 = NULL;
+    mps_transform_destroy(t1);
+    after(myrootExact, perset, 1, -10, 2, +10);
+
+    /* Two transforms, both live [-- not supported yet.  RHSK 2010-12-16] */
```

Comment:
It is true that multiple transforms are not supported yet?

Reply:
JPH investigate